scalalib/src/publish/ExtraPublish.scala

```diff
@@ -0,0 +1,25 @@
+package mill.scalalib.publish
+
+import mill.PathRef
+
+/** An extra resource artifact to publish.
+ *
+ * @param file The artifact file
+ * @param ivyType The ivy type, this will implicitly define the directory, the file will be published to (e.g. "jar" -> "jars")
+ *  @param ivyClassifier An Optional classifier to be used when publishing the file
+ *  @param ivyConfig The ivy config to be used
+ * @param ivyExt The extension that will be used publishing the file to the ivy repo
+ */
+case class ExtraPublish(
+  file: PathRef,
+  ivyType: String = "jar",
+  ivyClassifier: Option[String] = None,
+  ivyConfig: String = "compile",
+  ivyExt: String = "jar"
```

Comment:
The classifier, extentsion are also relevant for Maven repositories, so we should remove the "ivy" prefix.

What means the default value "compile" for `ivyConfig`. Does that mean, coursier will always resolve the artifact when compiling? That's not what I would expect for an additional resource like e.g. a distribution zip file.

Reply:
TBH. I have no idea. AFAIK the config can be defined by the user and "compile" seemed to be a reasonable default. We can use "runtime" instead, but then you have to remember to use the config when coding the dependenca in the mill build. At the end of the day you only reference the extra published artifact just when you want it resolved .